apps/studio/src/features/editing-experience/components/RootStateDrawer.tsx

```diff
@@ -24,20 +27,58 @@ export default function RootStateDrawer() {
     setSavedPageState,
     setPreviewPageState,
   } = useEditorDrawerContext()
+  const router = useRouter()
+  const pageId = Number(router.query.pageId)
+  const siteId = Number(router.query.siteId)
+  const { mutate } = trpc.page.reorderBlock.useMutation()
+  const toast = useToast({ status: "error" })
 
   const onDragEnd = (result: DropResult) => {
```

Comment:
nit: feels like a heavy function, may want to use `useCallback`

Reply:
assuming this is for `onDragEnd`